src/controllers/wishlistController.ts

```diff
@@ -0,0 +1,113 @@
+import { Request, Response } from 'express'
+import Product from '../database/models/productModel'
+import Wishlist from '../database/models/wishlistModel'
+
+/**
+ * Wishlist Controller class
+ */
+export default class WishlistController {
+  /**
+   * Adding product to wishlist
+   * @param {Request} req - Express request object
+   * @param {Response} res - Express response object
+   * @returns {Promise<Response>} Promise that resolves to an Express response
+   */
+  static async addToWishlist(req: Request, res: Response): Promise<Response> {
+    try {
+      const { productId } = req.body
+      const buyerId = req.user?.id
+
+      const product = await Product.findByPk(productId)
+      if (!product) {
+        return res.status(404).json({ message: 'Product not found' })
+      }
+
+      const wishlistExist = await Wishlist.findOne({
+        where: { productId, buyerId },
+      })
+      if (wishlistExist) {
+        return res.status(400).json({ message: 'Product already in wishlist' })
+      }
+
+      const addedWishlist = await Wishlist.create({
+        buyerId,
+        productId,
+      })
+
+      return res.status(201).json({
+        message: 'Product added to wishlist successfully',
+        wishlist: addedWishlist,
+        product: {
+          id: product.id,
+          name: product.name,
+          price: product.price,
+          imageUrl: product.images[2],
+        },
+      })
+    } catch (error) {
+      return res
+        .status(500)
+        .json({ message: 'Internal server error', error: error.message })
+    }

```

Comment:
```suggestion
    try {
      const { productId } = req.body
      const buyerId = req.user?.id

      const product = await Product.findByPk(productId)
      if (!product) {
        return res.status(404).json({ message: 'Product not found' })
      }

      const wishlistExist = await Wishlist.findOne({
        where: { productId, buyerId },
      })
      if (wishlistExist) {
        return res.status(400).json({ message: 'Product already in wishlist' })
      }

      const addedWishlist = await Wishlist.create({
        buyerId,
        productId,
      })

      return res.status(201).json({
        message: 'Product added to wishlist successfully',
        wishlist: addedWishlist,
        product: {
          id: product.id,
          name: product.name,
          price: product.price,
          imageUrl: product.images[2],
        },
      })
    } catch (error) {
      return res
        .status(500)
        .json({ message: 'Internal server error', error: error.message })
    }

```
this try and catch can be handled by error handler middleware, it is making the codes more verbose, it would be better it is removed